servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/DefaultGrpcClientMetadata.java

```diff
@@ -31,37 +32,94 @@
+     * strategy.
+     *
+     * @param path for the associated <a href="https://www.grpc.io">gRPC</a> method.
+     * @param requestEncoding {@link ContentCodec} to use for the associated <a href="https://www.grpc.io">gRPC</a>
+     * method.
+     * @param deadline A timeout deadline after which the response is no longer wanted.
+     */
+    protected DefaultGrpcClientMetadata(final String path,
+                                        final ContentCodec requestEncoding,
+                                        final Duration deadline) {
+        this(path, null, requestEncoding, deadline);
+    }
+
+    /**
+     * Creates a new instance which uses the provided path and execution strategy and the default content codec
+     * (identity) and no deadline.
      *
      * @param path for the associated <a href="https://www.grpc.io">gRPC</a> method.
      * @param strategy {@link GrpcExecutionStrategy} to use for the associated <a href="https://www.grpc.io">gRPC</a>
      * method.
      */
     protected DefaultGrpcClientMetadata(final String path,
                                         @Nullable final GrpcExecutionStrategy strategy) {
-        this(path, strategy, identity());
+        this(path, strategy, identity(), null);
     }
 
+    /**
+     * Creates a new instance which uses the provided path, content codec and the default execution
+     * strategy and no deadline.
+     *
+     * @param path for the associated <a href="https://www.grpc.io">gRPC</a> method.
+     * @param strategy {@link GrpcExecutionStrategy} to use for the associated <a href="https://www.grpc.io">gRPC</a>
+     * method.
+     * @param requestEncoding {@link ContentCodec} to use for the associated <a href="https://www.grpc.io">gRPC</a>
+     * method.
+     */
     protected DefaultGrpcClientMetadata(final String path,
                                         @Nullable final GrpcExecutionStrategy strategy,
                                         final ContentCodec requestEncoding) {
+        this(path, strategy, requestEncoding, null);
+    }
+
+    /**
+     * Creates a new instance which uses the provided path, content codec and deadline and the default execution
+     * strategy.
+     *
+     * @param path for the associated <a href="https://www.grpc.io">gRPC</a> method.
+     * @param strategy {@link GrpcExecutionStrategy} to use for the associated <a href="https://www.grpc.io">gRPC</a>
+     * method.
+     * @param requestEncoding {@link ContentCodec} to use for the associated <a href="https://www.grpc.io">gRPC</a>
+     * method.
+     * @param deadline A timeout deadline after which the response is no longer wanted.
+     */
+    protected DefaultGrpcClientMetadata(final String path,
+                                        @Nullable final GrpcExecutionStrategy strategy,
+                                        final ContentCodec requestEncoding,
+                                        @Nullable final Duration deadline) {
```

Comment:
Hm, we probably should think about a builder for meta-data. The list of params keeps growing. No necessary to do that as part of this PR, but worth considering.